src/e2e/java/uk/gov/hmcts/reform/sscs/functional/handlers/BaseHandler.java

```diff
@@ -44,26 +48,47 @@ public void setUp() {
         idamTokens = idamService.getIdamTokens();
     }
 
-    protected SscsCaseDetails createCaseInResponseReceivedState() throws Exception {
+    protected SscsCaseDetails createCaseInResponseReceivedState(int retry) throws Exception {
         SscsCaseDetails caseDetails = ccdService.createCase(buildSscsCaseDataForTesting("Mercury", "JK 77 33 22 Z"),
                 EventType.CREATE_TEST_CASE.getCcdType(), CREATED_BY_FUNCTIONAL_TEST,
                 CREATED_BY_FUNCTIONAL_TEST, idamTokens);
 
-        return ccdService.updateCase(caseDetails.getData(), caseDetails.getId(), EventType.DWP_RESPOND.getCcdType(),
+        return updateCase(retry, caseDetails.getData(), caseDetails.getId(), EventType.DWP_RESPOND.getCcdType(),
                 CREATED_BY_FUNCTIONAL_TEST, CREATED_BY_FUNCTIONAL_TEST, idamTokens);
     }
 
-    public SscsCaseDetails createCaseInWithDwpState() throws Exception {
+    protected SscsCaseDetails createCaseInWithDwpState(int retry) throws Exception {
         SscsCaseDetails caseDetails = ccdService.createCase(buildSscsCaseDataForTesting("Lennon", "BB 22 55 66 B"),
                 EventType.CREATE_TEST_CASE.getCcdType(), CREATED_BY_FUNCTIONAL_TEST,
                 CREATED_BY_FUNCTIONAL_TEST, idamTokens);
 
-        Thread.sleep(5000);
-
-        return ccdService.updateCase(caseDetails.getData(), caseDetails.getId(), EventType.SENT_TO_DWP.getCcdType(),
+        return updateCase(retry, caseDetails.getData(), caseDetails.getId(), EventType.SENT_TO_DWP.getCcdType(),
                 CREATED_BY_FUNCTIONAL_TEST, CREATED_BY_FUNCTIONAL_TEST, idamTokens);
     }
 
+    protected String getMyaResponse(int retry, Long caseId) {
+        Response response = null;
+        while (retry >= 0 && (response == null || response.statusCode() != HttpStatus.OK.value())) {
+            response = RestAssured
+                    .given()
+                    .when()
+                    .get("appeals?mya=true&caseId=" + caseId);
+        }
+
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.OK.value());
+        return response.then().extract().body().asString();
+    }
+
+    private SscsCaseDetails updateCase(int retry, SscsCaseData caseData, Long caseId, String eventType, String summary, String description, IdamTokens idamTokens) throws InterruptedException {
+        SscsCaseDetails caseDetails = null;
+        while (retry >= 0 && caseDetails == null) {
+            Thread.sleep(5000);
+            caseDetails = ccdService.updateCase(caseData, caseId, eventType, summary, description, idamTokens);
```

Comment:
Is caseDetails null when it fails to update? If so, can we add a log message so we can see when this happens?